src/jenkins.ts

```diff
@@ -0,0 +1,231 @@
+   */
+  readonly seedJobs?: SeedJob[];
+}
+
+/**
+ * A jenkins instance.
+ */
+export class Jenkins extends Construct {
+  private _plugins: Plugin[];
+  private _basePlugins: Plugin[];
+  private _seedJobs: SeedJob[];
+
+  constructor(scope: Construct, id: string, props: JenkinsProps = {}) {
+    super(scope, id);
+
+    this._plugins = props.plugins ?? [];
+    this._seedJobs = props.seedJobs ?? [];
+    this._basePlugins = [];
+    this.addBasePlugins(...(props.basePlugins ?? []));
+    const metadata = {
+      ...props.metadata,
+      namespace: props.metadata?.namespace ?? 'default',
+      labels: props.metadata?.labels ?? { app: 'jenkins' },
+    };
+    const authorizationStrategy = 'createUser';
+    const disableCSRFProtection = props.disableCsrfProtection ?? false;
+    // Keeping container resources defaults similar to Jenkins defaults here: https://jenkinsci.github.io/kubernetes-operator/docs/getting-started/latest/installing-the-operator/
+    const containers = [
+      {
+        name: 'jenkins-master',
+        image: 'jenkins/jenkins:lts',
+        imagePullPolicy: 'Always',
+        resources: {
+          limits: {
+            cpu: jenkins.JenkinsSpecMasterContainersResourcesLimits.fromString('1500m'),
+            memory: jenkins.JenkinsSpecMasterContainersResourcesLimits.fromString('3Gi'),
+          },
+          requests: {
+            cpu: jenkins.JenkinsSpecMasterContainersResourcesLimits.fromString('1'),
+            memory: jenkins.JenkinsSpecMasterContainersResourcesLimits.fromString('500M'),
+          },
+        },
+      },
+    ];;
+
+    const jenkinsCustomResource = new jenkins.Jenkins(this, 'Default', {
+      metadata: metadata,
+      spec: {
+        groovyScripts: {
+          configurations: [],
+          secret: {
+            name: '',
+          },
+        },
+        configurationAsCode: {
+          configurations: [],
+          secret: {
+            name: '',
+          },
+        },
+        jenkinsApiSettings: {
+          authorizationStrategy: authorizationStrategy,
+        },
+        master: {
+          disableCsrfProtection: disableCSRFProtection,
+          containers: containers,
+          basePlugins: cdk8s.Lazy.any({
+            produce: () => {
+              return this._basePlugins;
+            },
+          }),
+          plugins: cdk8s.Lazy.any({
+            produce: () => {
+              return this._plugins;
+            },
+          }),
+        },
+        seedJobs: cdk8s.Lazy.any({
+          produce: () => {
+            return this._seedJobs;
+          },
+        }),
+      },
+    });
+
+    this.node.defaultChild = jenkinsCustomResource;
+  }
+
+  /**
+   * Add base plugins to jenkins instance.
+   * @param basePlugins List of base plugins.
+   */
+  public addBasePlugins(...basePlugins: Plugin[]): void {
+    this._basePlugins.splice(0);
+
+    for (const basePlugin of basePlugins) {
+      DEFAULT_BASE_PLUGINS.set(basePlugin.name, basePlugin.version);
```

Comment:
Being a global constant, this will affect subsequent instances of `Jenkins`, that might not have configured the same overrides. Either copy the map and mutate the copy, or store it as an instance field.

Reply:
Ahh right, sorry I missed that. 🤦 Fixing it.

Reply:
I am curious how two jenkins instances manifest will look like. Will add a test for it.